npc/re/merchants/enchan_edda_half_moon.txt

```diff
@@ -0,0 +1,273 @@
+			.@part = EQI_ACC_R;
+			break;
+		case 3:
+			mes "Oh, why!!! I'm being nice to you!!!";
+			close;
+		}
+		.@itemid = getequipid(.@part);
+		if (.@itemid != 28387) {
+			mes "Uh, wait! That's not the ^0000cd<ITEM>"+getitemname(28387)+"<INFO>"+28387+"</INFO></ITEM>^000000, is it? Can you check it again?
+			close;
+		}
+		.@card = getequipcardid(.@part,3);
+		if (getiteminfo(.@card, ITEMINFO_SUBTYPE) == CARD_ENCHANT) {
+			npctalk "This necklace of yours already has an enchantment";
+			close;
+		}
+		mes "What special abilities would you like to enchant your necklace with?";
+		next;
+		.@menu$ = "";
+		setarray .@enhance_n$,"Goddess of Justice","Goddess of Mercy","Goddess of Insight";
+		for (.@i=0; .@i<getarraysize(.@enhance_n$); .@i++) {
+			.@menu$ += .@enhance_n$[.@i]+":";
+		}
+		.@menu$ += "Cancel";
+		.@select = select(.@menu$) -1;
+		if (.@select==getarraysize(.@enhance_n$)+1) {
+			callsub S_Cancel;
+		}
+		setarray .@enhance_a,29121,29123,29125;
+		setarray .@enhance_s,29122,29124,29126;
+		mes "[Temple Item Manager]";
+		mes "You have selected the "+.@enhance_n$[.@select]+" enchantment.";
+		mes "Please select the class";
+		mes "<ITEM>"+getitemname(.@enhance_a[.@select])+"<INFO>"+.@enhance_a[.@select]+"</INFO></ITEM>";
+		mes "<ITEM>"+getitemname(.@enhance_s[.@select])+"<INFO>"+.@enhance_s[.@select]+"</INFO></ITEM>";
+		next;
+		.@class = select("A Class:S Class");
+		.@enchant = .@class==1?.@enhance_a[.@select]:.@enhance_s[.@select];
+		.@cost = .@class==1?100:1000;
+		mes "[Temple Item Manager]";
+		mes "The necklace will be enchanted with the <ITEM>"+getitemname(.@enchant)+"<INFO>"+.@enchant+"</INFO></ITEM>";
+		mes "Are you really sure to proceed with this enchantment process?";
+		next;
+		if (select("No, I need time to consider.","Yes, please proceed.") == 1) {
+			callsub S_Cancel;
+		}
+		mes "[Temple Item Manager]";
+		if (!getequipisequiped(.@part) || isequipped(28387) < 1) {
+			mes "You aren't wearing the <ITEM>"+getitemname(28387)+"<INFO>"+28387+"</INFO></ITEM>...!";
+			close;
+		}
+		if (.@enchant == .@card) {
+			mes "Your <ITEM>"+getitemname(28387)+"<INFO>"+28387+"</INFO></ITEM> already has an enchant.";
+			close;
+		}
+		if (countitem(25088) < .@cost) {
+			mes "You didn't bring enough <ITEM>"+getitemname(25088)+"<INFO>"+25088+"</INFO></ITEM>!";
+			close;
+		}
+		if (.@class == 2) {
+			if (.@enchant == 29122 && .@card != 29121 ||.@enchant == 29124 && .@card != 29123 || .@enchant == 29126 && .@card != 29125 ) {
+				mes "You need to have your necklace enchanted with <ITEM>"+getitemname(.@enhance_a[.@select])+"<INFO>"+.@enhance_a[.@select]+"</INFO></ITEM> first.";
+				close;
+			}
+		}
+		specialeffect2 182;
+		specialeffect2 205;
+		specialeffect2 102;
+		specialeffect2 104;
+		mes "The necklace has been enchanted according to your specifications.";
+		delitem 25088,.@cost; //Fragment of Dream
+		delequip .@part;
+		getitem2 28387,1,1,0,0,0,0,0,.@enchant; //Priest's Necklace
+		close;
+	case 3:
+		S_Cancel;
```

Comment:
A doubt, why put a function in the case, to call it?
Wouldn't it be better to put the S_Cancel contents; inside this case 3?